include/seqan3/io/sam_file/format_bam.hpp

```diff
@@ -352,6 +354,15 @@ inline void format_bam::read_alignment_record(stream_type & stream,
 
             read_field(stream_view, tmp32); // l_ref (length of reference sequence)
 
+            // If there was no header text, we parse reference sequences block as header information
+            if (!header_present)
+            {
+                header.ref_ids().push_back(string_buffer);
+                header.ref_id_info.emplace_back(tmp32, "");
+                header.ref_dict[(header.ref_ids())[(header.ref_ids()).size() - 1]] = (header.ref_ids()).size() - 1;
```

Comment:
```suggestion
                auto & reference_ids = header.ref_ids();

                // put the length of the reference sequence into ref_id_info
                header.ref_id_info.emplace_back(tmp32, "");
                // put the reference name into reference_ids
                reference_ids.push_back(string_buffer);
                // assign the reference name an ascending reference id (starts at index 0).
                header.ref_dict.try_emplace(reference_ids.back(), reference_ids.size() - 1);
```

not sure if this is more readable?


```cpp
header.ref_id_info.emplace_back(tmp32, "");
```
I don't understand the data structure (std::tuple<int, std::string>) and the help page wasn't helpful either.

Reply:
I used `emplace` because it may move and duplicate entries are not allowed.

The `int` is the reference length, `string` is all tags that might be set in the header (`DS`, etc.)

Reply:
Okay, regarding the duplicate entries, I would have assumed that they shouldn't be duplicate in this case :)